### PR TITLE
Daemon: Port to GTK4

### LINKED ISSUE #1695: No context menu on wallpaper in Wayland
### What Happened?

I'm not able to reliably open the context menu when running under Wayland

### Steps to Reproduce

1. Run a wayland session
2. secondary click the wallpaper
3. Nothing happens

### Expected Behavior

Context menu with options for changing the wallpaper, display settings, etc

### OS Version

7.x (Horus)

### Software Version

Compiled from git

### Log Output

_No response_

### Hardware Info

_No response_

### LINKED ISSUE #1696: Window context menu doesn't automatically close on Wayland
### What Happened?

The context menu that appears when secondary clicking on window titles doesn't close when clicking outside of it

### Steps to Reproduce

1. Run a wayland session
2. Secondary click on a window title
3. See the context menu with window management options
4. Click outside of the menu to dismiss it
5. nothing happens

### Expected Behavior

Clicking outside of the menu should dismiss it

### OS Version

7.x (Horus)

### Software Version

Compiled from git

### Log Output

_No response_

### Hardware Info

_No response_

### PR DESCRIPTION
**Pull request by leolost2605**
Fixes #1696
Fixes #1695

### CODE REVIEW

File: daemon/WindowMenu.vala [hunk 1]
```diff
@@ -3,11 +3,13 @@
...
  */
 
-public class Gala.Daemon.WindowMenu : Gtk.Menu {
+public class Gala.Daemon.WindowMenu : Gtk.Popover {
     private static GLib.Settings gala_keybind_settings = new GLib.Settings ("org.pantheon.desktop.gala.keybindings");
     private static GLib.Settings keybind_settings = new GLib.Settings ("org.gnome.desktop.wm.keybindings");
 
-    public signal void perform_action (Gala.ActionType type);
+    public signal void perform_action (Gala.ActionType type) {
+        popdown ();
```
Comment (danirabbit) [inline daemon/WindowMenu.vala:11]: I didn't even know you could have a function attached to a signal. That's dope

File: meson.build [hunk 2]
```diff
@@ -89,10 +87,11 @@ gio_dep = dependency('gio-2.0', version: '>= @0@'.format(glib_version_required))
 gio_unix_dep = dependency('gio-unix-2.0', version: '>= @0@'.format(glib_version_required))
 gmodule_dep = dependency('gmodule-2.0')
 gtk_dep = [dependency('gtk+-3.0', version: '>= @0@'.format(gtk_version_required)), dependency('gdk-x11-3.0')]
+gtk4_dep = dependency('gtk4')
 gee_dep = dependency('gee-0.8')
-granite_dep = dependency('granite', version: '>= 5.4.0')
+granite_dep = dependency('granite-7')
 gnome_desktop_dep = dependency('gnome-desktop-3.0')
-hdy_dep = dependency('libhandy-1')
+adw_dep = dependency('libadwaita-1')
```
Comment (danirabbit) [inline meson.build:94]: Are you actually using Adw here? I think we were only using Hdy for the window

Comment (leolost2605) [reply meson.build]: Ah right :)

**Comment by leolost2605** [comment]
Oh right X is a thing oops :sweat_smile: 
I think monitor labels won't work at all on X :(
Maybe we should keep the gtk3 version as legacy and depending on wayland or x launch the right one? Shouldn't be a maintenance burden since we probably don't have to fix anything there? And we always can just drop it :shrug:

**Comment by danirabbit** [comment]
@leolost2605 that makes sense to me! If that's not too much trouble for you

**Comment by leolost2605** [comment]
@danirabbit done!

**Pull request merged**